File: app/src/main/java/com/numero/sojodia/SojoDiaApplication.java
package com.numero.sojodia;

import android.app.Application;

import com.numero.sojodia.di.ApiModule;
import com.numero.sojodia.di.ApplicationComponent;
import com.numero.sojodia.di.ApplicationModule;
import com.numero.sojodia.di.RepositoryModule;
import com.numero.sojodia.di.DaggerApplicationComponent;

public class SojoDiaApplication extends Application {

    private ApplicationComponent component;

    @Override
    public void onCreate() {
        super.onCreate();

        component = DaggerApplicationComponent.builder()
                .applicationModule(new ApplicationModule(this))
                .repositoryModule(new RepositoryModule())
                .apiModule(new ApiModule())
                .build();
    }

    public ApplicationComponent getComponent() {
        return component;
    }
}

File: app/src/main/kotlin/com/numero/sojodia/SojoDiaApplication.kt
package com.numero.sojodia

import android.app.Application
import com.numero.sojodia.extension.applyApplication
import com.numero.sojodia.repository.BusDataRepository
import com.numero.sojodia.repository.BusDataRepositoryImpl
import com.numero.sojodia.repository.ConfigRepository
import com.numero.sojodia.repository.ConfigRepositoryImpl

class SojoDiaApplication : Application(), Module {

    override lateinit var configRepository: ConfigRepository

    override lateinit var busDataRepository: BusDataRepository

    override fun onCreate() {
        super.onCreate()
        configRepository = ConfigRepositoryImpl(this)

        busDataRepository = BusDataRepositoryImpl(this)
        configRepository.appTheme.applyApplication()
    }
}
